src/app/components/volume-knob/volume-knob.directive.js

```diff
@@ -0,0 +1,81 @@
+class VolumeKnobDirective {
```

Comment:
This class is a clone of knob.directive, if you want to decorate / extend the knob directive you should use that one, here you are duplicating a lot of code. The scope of knob.directive is to be a generic / reusable knob with the behaviour of debounce / clusterize the model changes. I don't see specific requirements for the volume to create a new specialised component.

To test your changes open 3 UI binded to the same PI in different windows and start to play with the volume